app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -71,18 +75,32 @@ public void run() {
 
     private void receiveLevelDetails() {
         // TODO: Serialize current level data and put it in some container
+
         levelTimePoints = 200;
-        levelSpeed = 50;
+        levelSpeed = levelData.getBaseSpeed();
     }
 
     public void updateGameState() {
         // TODO: Check if time point is in level data and add data to some container, than move
         //  it to left
+        checkTimePoint();
         updateDroidCoordinates();
+    }
+
+    private void checkTimePoint() {
+        if (levelData.isEmpty()) {
 
-        // Level Finishing
-        if (timePoint == levelTimePoints) {
+            // When the obstacles end - the level is considered passed
             winGame();
```

Comment:
Add "return;" after "winGame" - this way you won't have to add "else" block.

Reply:
Done